datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -262,25 +262,25 @@ pub(crate) struct AggregationOrdering {
 #[derive(Debug)]
 pub struct AggregateExec {
     /// Aggregation mode (full, partial)
-    pub(crate) mode: AggregateMode,
```

Comment:
I'm not sure about this, do we want to expose the internals of a physical operator externally?

Reply:
I will improve the API

Reply:
> I'm not sure about this, do we want/need to expose the internals of a physical operator externally?

The short answer is there are things in the physical optimizer passes that use these fields directly (I think mostly as a convenience). I will make a real API